kover-maven-plugin/src/main/kotlin/kotlinx/kover/maven/plugin/mojo/abstracts/AbstractCoverageTaskMojo.kt

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2017-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.kover.maven.plugin.mojo.abstracts
+
+import kotlinx.kover.features.jvm.ClassFilters
+import kotlinx.kover.features.jvm.KoverFeatures
+import kotlinx.kover.maven.plugin.Constants
+import kotlinx.kover.maven.plugin.MavenReportFilters
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion
+import org.apache.maven.artifact.versioning.VersionRange
+import org.apache.maven.plugins.annotations.Parameter
+import org.apache.maven.project.MavenProject
+import org.codehaus.plexus.util.xml.Xpp3Dom
+import java.io.File
+import java.nio.file.Files
+import kotlin.io.path.Path
+
+/**
+ * Common class using for processing project coverage.
+ */
+abstract class AbstractCoverageTaskMojo : AbstractKoverMojo() {
+    /**
+     * Filters to limit the code that gets into the report.
+     */
+    @Parameter
+    private var filters: MavenReportFilters? = null
+
+    /**
+     * Flag to use code coverage from dependencies.
+     *
+     * `false` by default.
+     */
+    @Parameter(property = "kover.aggregate", defaultValue = "false")
+    protected var aggregate: Boolean = false
+
+    /**
+     * Binary reports that built in advance, before the start of the project build.
+     *
+     * It's convenient to use when tests are performed on CI on different nodes.
```

Comment:
It is worth at least a cursory read KDoc.
Additional files are provided only when necessary. That is, you need to merge the coverage with tests that are run on different machines. 
Since the XML or HTML report cannot be merged, only ic files can be used for merging.